lightning/src/ln/functional_tests.rs

```diff
@@ -3668,6 +3673,17 @@ fn do_test_drop_messages_peer_disconnect(messages_delivered: u8) {
 	nodes[0].node.peer_disconnected(&nodes[1].node.get_our_node_id(), false);
 	nodes[1].node.peer_disconnected(&nodes[0].node.get_our_node_id(), false);
 	if messages_delivered < 3 {
+		if simulate_broken_lnd {
+			// lnd has a long-standing bug where they send a funding_locked prior to a
+			// channel_reestablish if you reconnect prior to funding_locked time.
+			//
+			// Here we simulate that behavior, delivering a funding_locked immediately on
+			// reconnect. Note that we don't bother skipping the now-duplicate funding_locked sent
+			// in `reconnect_nodes` but we currently don't fail based on that.
+			//
+			// See-also <https://github.com/lightningnetwork/lnd/issues/4006>
+			nodes[1].node.handle_funding_locked(&nodes[0].node.get_our_node_id(), &as_funding_locked.as_ref().unwrap().0);
+		}
 		// Even if the funding_locked messages get exchanged, as long as nothing further was
 		// received on either side, both sides will need to resend them.
 		reconnect_nodes(&nodes[0], &nodes[1], (true, true), (0, 1), (0, 0), (0, 0), (0, 0), (false, false));
```

Comment:
Just a Q: since `send_funding_locked` is `(true, true)` here, wouldn't that result in a second nodes[0] funding_locked being sent? Is that similar enough to the lnd scenario we're trying to test? :thinking:

Reply:
Indeed, it will, I think its close enough, but I'm happy to rework if you disagree, its just a good chunk more special-case code here.

Reply:
I think it's OK. The test does fail with the previous code